src/cloud-api-adaptor/pkg/podnetwork/tunneler/routing/podnode.go

```diff
@@ -39,6 +38,7 @@ func (t *podNodeTunneler) Setup(nsPath string, podNodeIPs []netip.Addr, config *
 		return errors.New("secondary pod node IP is not available")
 	}
 
+	podVEthName := config.InterfaceName
```

Comment:
Have you considered the cases where `config.InterfaceName` is not defined? Should we have a default value? Or maybe a simpler approach for now where we check if `podVEthName` is valid the same way we are doing on line 47 for `hostNS`?

Reply:
Yes. we need an empty string check here. I'll add it.

Reply:
done